### ISSUE TITLE
SMB2 gives NT_STATUS_OBJECT_NAME_NOT_FOUND, SMB1 works fine (Samba NAS)

### ISSUE THREAD

**Issue by dkewley** [yoichiro/chromeos-filesystem-cifs#114]
Using extension version 2.0.4 to access my Synology NAS (running Samba 4.1.18, Synology DSM 5.2.0.5592.4), SMB2 fails while SMB1/CIFS works. Because I can limit the protocol to SMB1/CIFS in the extension's configuration dialog box (the gear icon), I can get my work done using SMB1. I thought you'd want to know I ran into this issue, which others may see as well.

Using SMB2, I get the following output on the console, ending with the request for \srvsvc, and an error of NT_STATUS_OBJECT_NAME_NOT_FOUND:

```
Looked up with chrome.mdns:  Array[0]
background.js:18387 Looked up with chrome.mdns:  Array[1]
background.js:18387 Looked up with chrome.mdns:  Array[1]
background.js:6379 Domain name not specified. '?' will be applied.
background.js:6372 connect
background.js:6372 Header {types_: Types, protocol_: "SMB", command_: 114, errorClass_: 0, errorCode_: 0…}
background.js:6372 NegotiateProtocolRequest {types_: Types, dialects: Array[0], dialects_: Array[2]}dialects: Array[0]dialects_: Array[2]types_: Types__proto__: NegotiateProtocolRequest
background.js:6372 Packet {types_: Types, data_: ArrayBuffer, dataLength_: 58, packetHelper_: PacketHelper}
background.js:6379 Negotiated with SMB2
background.js:6372 Header {types_: Types, protocol_: "SMB", structureSize_: 64, creditCharge_: 0, channelSequence_: 0…}
background.js:6372 NegotiateResponse {types_: Types, structureSize_: 65, securityMode_: 1, dialectRevision_: 514, serverGuid_: "diskstation"…}
background.js:6372 Header {types_: Types, protocol_: "SMB", structureSize_: 64, creditCharge_: 1, channelSequence_: 0…}
background.js:6372 SessionSetupRequest {types_: Types, flags_: 0, securityMode_: 1, capabilities_: 1, channel_: 0…}
background.js:6372 Packet {types_: Types, data_: ArrayBuffer, dataLength_: 129, packetHelper_: PacketHelper}
background.js:6372 Header {types_: Types, protocol_: "SMB", structureSize_: 64, creditCharge_: 1, channelSequence_: 22…}
background.js:6379 NTLMSSP_NEGOTIATE_UNICODE: true
background.js:6379 NTLMSSP_NEGOTIATE_OEM: false
background.js:6379 NTLMSSP_REQUEST_TARGET: true
background.js:6379 NTLMSSP_NEGOTIATE_SIGN: false
background.js:6379 NTLMSSP_NEGOTIATE_SEAL: false
background.js:6379 NTLMSSP_NEGOTIATE_DATAGRAM_STYLE: false
background.js:6379 NTLMSSP_NEGOTIATE_LM_KEY: false
background.js:6379 NTLMSSP_NEGOTIATE_NETWARE: false
background.js:6379 NTLMSSP_NEGOTIATE_NTLM: true
background.js:6379 NTLMSSP_NEGOTIATE_OEM_DOMAIN_SUPPLIED: false
background.js:6379 NTLMSSP_NEGOTIATE_OEM_WORKSTATION_SUPPLIED: false
background.js:6379 NTLMSSP_NEGOTIATE_LOCAL_CALL: false
background.js:6379 NTLMSSP_NEGOTIATE_ALWAYS_SIGN: false
background.js:6379 NTLMSSP_TARGET_TYPE_DOMAIN: false
background.js:6379 NTLMSSP_TARGET_TYPE_SERVER: true
background.js:6379 NTLMSSP_TARGET_TYPE_SHARE: false
background.js:6379 NTLMSSP_NEGOTIATE_NTLM2: true
background.js:6379 NTLMSSP_REQUEST_INIT_RESPONSE: false
background.js:6379 NTLMSSP_REQUEST_ACCEPT_RESPONSE: false
background.js:6379 NTLMSSP_REQUEST_NON_NT_SESSION_KEY: false
background.js:6379 NTLMSSP_NEGOTIATE_TARGET_INFO: true
background.js:6379 NTLMSSP_NEGOTIATE_128: true
background.js:6379 NTLMSSP_NEGOTIATE_KEY_EXCH: false
background.js:6379 NTLMSSP_NEGOTIATE_56: false
background.js:6372 Header {types_: Types, protocol_: "SMB", structureSize_: 64, creditCharge_: 1, channelSequence_: 0…}
background.js:6372 SessionSetupRequest {types_: Types, flags_: 0, securityMode_: 1, capabilities_: 1, channel_: 0…}
background.js:6372 Packet {types_: Types, data_: ArrayBuffer, dataLength_: 330, packetHelper_: PacketHelper}
background.js:6372 Header {types_: Types, protocol_: "SMB", structureSize_: 64, creditCharge_: 1, channelSequence_: 0…}
background.js:6372 TreeConnectRequest {types_: Types, path_: "\\192.168.2.10\IPC$"}
background.js:6372 Packet {types_: Types, data_: ArrayBuffer, dataLength_: 110, packetHelper_: PacketHelper}
background.js:6372 Header {types_: Types, protocol_: "SMB", structureSize_: 64, creditCharge_: 1, channelSequence_: 0…}
background.js:6372 TreeConnectResponse {types_: Types, structureSize_: 16, shareType_: 2, shareFlags_: 0, capabilities_: 0…}
background.js:6372 Header {types_: Types, protocol_: "SMB", structureSize_: 64, creditCharge_: 1, channelSequence_: 0…}
background.js:6372 CreateRequest {types_: Types, structureSize_: 57, requestedOplockLevel_: 0, impersonationLevel_: 2, desiredAccess_: -1073741824…}createContexts_: Array[1]createDisposition_: 1createOptions_: 0desiredAccess_: -1073741824fileAttributes_: 128impersonationLevel_: 2name_: "\srvsvc"requestedOplockLevel_: 0shareAccess_: 3structureSize_: 57types_: Types__proto__: CreateRequest
background.js:6372 Packet {types_: Types, data_: ArrayBuffer, dataLength_: 160, packetHelper_: PacketHelper}
background.js:15801 c0000034: NT_STATUS_OBJECT_NAME_NOT_FOUND
```

Using SMB (forced by limiting the version in the extension's config dialog box), I get:

```
...
Negotiated with SMB1/CIFS
...
TreeConnectAndxRequest {types_: Types, path_: "\\192.168.2.10\IPC$", service_: "?????"}
background.js:6372 Packet {types_: Types, data_: ArrayBuffer, dataLength_: 90, packetHelper_: PacketHelper}
background.js:6372 Header {types_: Types, protocol_: "SMB", command_: 117, errorClass_: 0, errorCode_: 0…}
background.js:6372 TreeConnectAndxResponse {types_: Types, responseUtils_: ResponseUtils, supportSearchBits_: true, shareIsInDfs_: false, service_: "IPC"}responseUtils_: ResponseUtilsservice_: "IPC"shareIsInDfs_: falsesupportSearchBits_: truetypes_: Types__proto__: TreeConnectAndxResponse
background.js:6372 Header {types_: Types, protocol_: "SMB", command_: 162, errorClass_: 0, errorCode_: 0…}
background.js:6372 NtCreateAndxRequest {types_: Types, flags_: 22, rootDirectoryFId: 0, desiredAccess_: 131487, allocationSize_: 0…}allocationSize_: 0createDisposition_: 1createOptions_: 64desiredAccess_: 131487extFileAttributes_: 128fileName_: "\srvsvc"flags_: 22impersonationLevel_: 2rootDirectoryFId: 0rootDirectoryFId_: 0securityFlags_: 3shareAccess_: 7types_: Types__proto__: NtCreateAndxRequest
background.js:6372 Packet {types_: Types, data_: ArrayBuffer, dataLength_: 100, packetHelper_: PacketHelper}
background.js:6372 Header {types_: Types, protocol_: "SMB", command_: 162, errorClass_: 0, errorCode_: 0…}
background.js:6372 NtCreateAndxResponse {types_: Types, responseUtils_: ResponseUtils, opLockLevel_: 0, fid: null, createDisposition_: 1…}allocationSize_: 0createDisposition_: 1createTime_: Sun Dec 31 1600 16:00:00 GMT-0800 (PST)directory_: 0endOfFile_: 0extFileAttributes_: 128fid: nullfid_: 53306lastAccessTime_: Sun Dec 31 1600 16:00:00 GMT-0800 (PST)lastChangeTime_: Sun Dec 31 1600 16:00:00 GMT-0800 (PST)lastWriteTime_: Sun Dec 31 1600 16:00:00 GMT-0800 (PST)nmPipeStatus_: 1535opLockLevel_: 0resourceType_: 2responseUtils_: ResponseUtilstypes_: Types__proto__: NtCreateAndxResponse
background.js:6372 Header {types_: Types, protocol_: "SMB", command_: 37, errorClass_: 0, errorCode_: 0…}
...
```

I've shown all console output lines for SMB2, but shown only the lines that I thought are relevant for SMB1. Let me know if you need more info.

**Comment by yoichiro** [comment]
@dkewley I just released a new version 2.1.1. The previous version included a bug which a wrong path specified (\srvsvc). I guess that you can access to your server with the latest version. It would be great if you can do test again.

**Comment by dkewley** [comment]
SMB2 works for me now in v2.1.1. I see the name was changed from `\srvsvc` to simply `srvsvc`. Thank you @yoichiro for the fix and quick response! Closing this issue.
